app/static/src/app/store/graphs/getters.ts

```diff
@@ -24,5 +27,12 @@ export const getters: GraphsGetters & GetterTree<GraphsState, AppState> = {
     [GraphsGetter.hiddenVariables]: (_, graphsGetters: GraphsGettersValues, rootState: AppState): string[] => {
         const allSelected = graphsGetters[GraphsGetter.allSelectedVariables];
         return rootState.model.odinModelResponse?.metadata?.variables.filter((s) => !allSelected.includes(s)) || [];
+    },
+    [GraphsGetter.legendWidth]: (_, graphsGetters): number => {
+        // Heuristic for setting graph legend width based on string length of longest variable name
+        const longestVar = graphsGetters[GraphsGetter.allSelectedVariables].sort((a: string, b: string) =>
+            a.length < b.length ? 1 : -1
+        )[0];
+        return longestVar.length * 10 + 40;
```

Comment:
would be get these names as variables

Reply:
Do you mean unmagic these numbers? Will do.